### PR TITLE
Update nginx connecto to 3.0.4

### PR DESCRIPTION
**Pull request by fzipi**
Also fixed some hadolint warnings in Dockerfile

Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

### CODE REVIEW

File: v3-nginx/Dockerfile [hunk 1]
```diff
@@ -72,7 +74,10 @@ RUN apt-get update \
       libxml2-dev \
       make \
       wget \
-      zlib1g-dev
+      zlib1g-dev \
+      vim-tiny && \
```
Comment (bittner) [inline v3-nginx/Dockerfile]: Do we need an editor in a productive WAF image?

File: v3-nginx/Dockerfile [hunk 2]
```diff
@@ -84,34 +89,33 @@ COPY --from=build /usr/share/TLS/server.key /etc/nginx/conf/server.key
...
     make modules && \
     cp objs/ngx_http_modsecurity_module.so /etc/nginx/modules
 
 RUN mkdir /etc/modsecurity.d && \
-    cd /etc/modsecurity.d && \
-    echo "include \"/etc/modsecurity.d/modsecurity.conf\"" > setup.conf && \
-    wget --quiet https://raw.githubusercontent.com/SpiderLabs/ModSecurity/v3/master/modsecurity.conf-recommended && \
-    mv modsecurity.conf-recommended modsecurity.conf && \
-    wget --quiet https://raw.githubusercontent.com/SpiderLabs/ModSecurity/v3/master/unicode.mapping
+    echo "include \"/etc/modsecurity.d/modsecurity.conf\"" > /etc/modsecurity.d/setup.conf && \
```
Comment (bittner) [inline v3-nginx/Dockerfile]: How about?
```suggestion
    echo 'include "/etc/modsecurity.d/modsecurity.conf"' > /etc/modsecurity.d/setup.conf && \
```
or even
```suggestion
    echo "include /etc/modsecurity.d/modsecurity.conf" > /etc/modsecurity.d/setup.conf && \
```

File: v3-nginx/Dockerfile [hunk 3]
```diff
@@ -84,34 +89,33 @@ COPY --from=build /usr/share/TLS/server.key /etc/nginx/conf/server.key
...
+      -o /etc/modsecurity.d/unicode.mapping
 
 RUN sed -i '1iload_module modules/ngx_http_modsecurity_module.so;' /etc/nginx/nginx.conf && \
     sed -i -e 's/http {/http {\n    modsecurity on;\n    modsecurity_rules_file \/etc\/modsecurity.d\/setup.conf;\n/g' /etc/nginx/nginx.conf
 
 ENV LD_LIBRARY_PATH /lib:/usr/lib:/usr/local/lib
 
-RUN apt-get install -y vim
-RUN echo $SERVERNAME
+RUN echo "$SERVERNAME"
```
Comment (bittner) [inline v3-nginx/Dockerfile]: I'm not sure I understand this line. That would add an image layer just for printing out the `SERVERNAME` env variable? Does that make sense?

File: v3-nginx/Dockerfile [hunk 4]
```diff
@@ -84,34 +89,33 @@ COPY --from=build /usr/share/TLS/server.key /etc/nginx/conf/server.key
...
 RUN sed -i -E "s/server_name .*;/server_name $SERVERNAME;/g" /etc/nginx/conf.d/default.conf
 
 RUN if [ "$SETTLS" = "True" ]; then \
       echo "setting TLS"; \
       mv /etc/nginx/conf.d/tls.conf.disabled /etc/nginx/conf.d/tls.conf; \
-      sed -i -E 's/server_name .*;/server_name $SERVERNAME;/g' /etc/nginx/conf.d/tls.conf; \
+      sed -i -E "s/server_name .*;/server_name $SERVERNAME;/g" /etc/nginx/conf.d/tls.conf; \
```
Comment (bittner) [inline v3-nginx/Dockerfile]: Using double-quotes will potentially allow to expand the globbing (`.*`) in the string. Is that intended?

File: v3-nginx/Dockerfile (hunk 1 above)
Comment (fzipi) [reply v3-nginx/Dockerfile]: No, but it was there before :) We can get rid of that one.

File: v3-nginx/Dockerfile (hunk 2 above)
Comment (fzipi) [reply v3-nginx/Dockerfile]: Excellent.

File: v3-nginx/Dockerfile (hunk 3 above)
Comment (fzipi) [reply v3-nginx/Dockerfile]: No, but it was there :) Removing.

File: v3-nginx/Dockerfile (hunk 4 above)
Comment (fzipi) [reply v3-nginx/Dockerfile]: Well, without quoting it won't work as intended:
```bash
[fzipi@railgun-lan example]$ cat nginx.conf 
server_name example.com;
[fzipi@railgun-lan example]$ SERVERNAME=globbing.com
[fzipi@railgun-lan example]$ sed -E 's/server_name .*;/server_name $SERVERNAME;/g' nginx.conf 
server_name $SERVERNAME;
[fzipi@railgun-lan example]$ sed -E "s/server_name .*;/server_name $SERVERNAME;/g" nginx.conf
server_name globbing.com;
```
You can do also any of:
```bash
[fzipi@railgun-lan example]$ sed -E "s/server_name .*;/server_name "$SERVERNAME";/g" nginx.conf 
server_name globbing.com;
[fzipi@railgun-lan example]$ sed -E "s/server_name .*;/server_name "${SERVERNAME}";/g" nginx.conf 
server_name globbing.com;
```

Comment (bittner) [reply v3-nginx/Dockerfile]: True. The double-quotes are needed for expanding the `$SERVERNAME`.

File: .github/workflows/dockerimage.yml [hunk 5]
```diff
@@ -8,15 +9,17 @@ jobs:
...
     - uses: actions/checkout@v1
       with:
         fetch-depth: 1
 
-    - name: Log in to Docker Hub
-      run: docker login
-           -u ${{ secrets.dockerhub_user }}
-           -p ${{ secrets.dockerhub_token }}
+    - uses: azure/docker-login@v1
+      if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```
Comment (bittner) [inline .github/workflows/dockerimage.yml]: The event name is probably unneeded. Note the `on: push` on line 2 above. Do you see any issue with that?

File: .github/workflows/dockerimage.yml [hunk 6]
```diff
@@ -25,5 +28,5 @@ jobs:
            --tag owasp/modsecurity:$(./version.sh ${{ matrix.image }} -vvv)
 
     - name: Push ${{ matrix.image }}
-      if: github.ref == 'refs/heads/master'
+      if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```
Comment (bittner) [inline .github/workflows/dockerimage.yml]: Same here. Note that the use of `if` (in our implementation here) is really just a workaround; it's procedural, but we should really favor a declarative implementation style (that's what `on: push` is about).

File: .github/workflows/dockerimage.yml (hunk 5 above)
Comment (fzipi) [reply .github/workflows/dockerimage.yml]: You are exactly right! (bad copy and paste)

File: .github/workflows/dockerimage.yml (hunk 6 above)
Comment (fzipi) [reply .github/workflows/dockerimage.yml]: You are exactly right! (bad copy and paste)

**Comment by fzipi** [comment]
@bittner Is there anything remaining here?

**Comment by fzipi** [comment]
Thanks!

**Comment by fzipi** [comment]
@bittner BTW, do you want an update to the README.md with the changed versions? I guess it should be 3.0.4 now?

**Comment by bittner** [comment]
Oh yeah, that slipped through. Would be good.

**Comment by bittner** [comment]
Wait! That's fine. I'll do it myself just now.

**Comment by bittner** [comment]
Thank you, @fzipi!

**Pull request merged**